### PR TITLE
Remove unused Connect dependencies 

### PR DESCRIPTION
**Pull request by gzdunek**
The removed deps are either unused or duplicated in other packages.

I also removed the `start-electron` script, I think no one uses it, we start the app with `start`. Let me know if I'm wrong.

I tested `start-term`, `build-term`, `start-teleport`, `build-teleport`, `storybook`, `lint` scripts.

### CODE REVIEW

File: web/packages/teleterm/package.json [hunk 1]
```diff
@@ -44,18 +43,14 @@
     "@types/node-forge": "^1.0.4",
     "@types/tar-fs": "^2.0.1",
     "@types/whatwg-url": "^11.0.1",
-    "clean-webpack-plugin": "4.0.0",
```
Comment (gzdunek) [inline web/packages/teleterm/package.json:47]: Duplicated in build/package.json, used.

File: web/packages/teleterm/package.json [hunk 2]
```diff
@@ -44,18 +43,14 @@
...
     "electron": "29.1.4",
     "electron-notarize": "^1.2.1",
     "electron-vite": "^2.0.0",
-    "eslint-import-resolver-webpack": "0.13.2",
```
Comment (gzdunek) [inline web/packages/teleterm/package.json:51]: Duplicated in `build/package.json`, but not used. I removed it from both places.

File: web/packages/teleterm/package.json [hunk 3]
```diff
@@ -44,18 +43,14 @@
...
-    "eslint-loader": "3.0.3",
```
Comment (gzdunek) [inline web/packages/teleterm/package.json:52]: No usages.

File: web/packages/teleterm/package.json [hunk 4]
```diff
@@ -44,18 +43,14 @@
...
     "google-protobuf": "^3.21.2",
     "immer": "^9.0.7",
     "jest-canvas-mock": "^2.3.1",
     "react-dnd": "^14.0.4",
     "react-dnd-html5-backend": "^14.0.2",
-    "ts-loader": "^9.4.2",
```
Comment (gzdunek) [inline web/packages/teleterm/package.json:58]: Duplicated in `teleport/package.json`, used.

**Review by ryanclark — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by ravicious — APPROVED** [review]

**Pull request merged**